### PR TITLE
Move the bigchaindb/nginx_3scale repo under bigchaindb/bigchaindb

### PR DESCRIPTION
**Pull request by muawiakh**
- All files moved to k8s/nginx-3scale with directory structure consistent
with k8s/nginx-http(s)
- Top level LICENCES.md updated
- Renaming entry point script to nginx_3scale_entrypoint.bash

### CODE REVIEW

**Comment by muawiakh** [comment]
partially fixes #1588

**Comment by krish7919** [comment]
We should ideally move it under `nginx-openresty/container` directory.

**Comment by muawiakh** [comment]
@krish7919 +1 on moving it under `nginx-openresty`. Done. I tried to do an initial commit for this and we can iterate once @ttmc is back?
Not sure about how we can conserve the history.
I am not clear on what impact would the `nginx-openresty-dep.yaml` file have with the name change?

**Comment by krish7919** [comment]
> Not sure about how we can conserve the history.

If we cannot preserve the history here, then I would lean on _not_ deleting the old repo as we might need to refer to it later.

> I am not clear on what impact would the nginx-openresty-dep.yaml file have with the name change?

Ideally, the yaml file will not change as it has no relation to the container internals; it pulls from Docker Hub and all it needs is the correct image and tag to be hosted there.

The changes look good to me. I would suggest to let it be open till @ttmc's review on licensing changes.

**Comment by ttmc** [comment]
I made some edits to the two `LICENSES.md` files; see commit https://github.com/bigchaindb/bigchaindb/pull/1728/commits/310433407cd9a0ab6d4a905d769cd31f3c960971

File: k8s/nginx-openresty/container/README.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
++-----------------------------+----+    |     Upstream API Port       |                        | Host       |
+|      MongoDB Frontend Port       |    |                             |                        |            |
+|                                  |    |[internal port where we can  |   +--------------------+--------+   |
+|  [port number for MongoDB        |    |access backend BDB cluster   |   |  MongoDB Backend Port       |   |
+|  instances to communicate        |    +--+--------------------------+   |                             |   |
+|  with each other                 +-------+----------------------------->|[port where MongoDB instance |   |
++-----------------------------+----+       |                              | listens/waits for requests] |   |
+                              |            |                              +--------------------+--------+   |
+                              +------------+                                                   +------------+
+```
```
Comment (krish7919) [inline k8s/nginx-openresty/container/README.md]: I guess the whole ASCII diagram can go now.

File: k8s/nginx-openresty/container/docker_build_and_push.bash [hunk 2]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+
+docker build -t bigchaindb/nginx_3scale:3.0 .
+
+docker push bigchaindb/nginx_3scale:3.0
```
Comment (krish7919) [inline k8s/nginx-openresty/container/docker_build_and_push.bash]: Let's bump the version to 3.1 to refer to the new repo from now on?

File: k8s/nginx-openresty/container/nginx_openresty_entrypoint.bash [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+sed -i "s|BIGCHAINDB_API_PORT|${bdb_api_port}|g" ${NGINX_CONF_FILE}
+sed -i "s|THREESCALE_RESPONSE_SECRET_TOKEN|${threescale_secret_token}|g" $NGINX_CONF_FILE
+sed -i "s|SERVICE_ID|${threescale_service_id}|g" $NGINX_CONF_FILE
+sed -i "s|THREESCALE_VERSION_HEADER|${threescale_version_header}|g" $NGINX_CONF_FILE
+sed -i "s|SERVICE_TOKEN|${threescale_service_token}|g" $NGINX_CONF_FILE
+
+
+# start nginx
+echo "INFO: starting nginx..."
+exec /usr/local/openresty/nginx/sbin/nginx -c ${NGINX_CONF_FILE}
```
Comment (krish7919) [inline k8s/nginx-openresty/container/nginx_openresty_entrypoint.bash]: Just check once that the conf and lua files are the latest from the nginx_3scale repo as this is an old PR and we do not want to miss any changes made there.

File: k8s/nginx-openresty/container/docker_build_and_push.bash (hunk 2 above)
Comment (muawiakh) [reply k8s/nginx-openresty/container/docker_build_and_push.bash]: Sure, I can do that.

File: k8s/nginx-openresty/container/README.md (hunk 1 above)
Comment (muawiakh) [reply k8s/nginx-openresty/container/README.md]: Done

File: k8s/nginx-openresty/container/nginx_openresty_entrypoint.bash (hunk 3 above)
Comment (muawiakh) [reply k8s/nginx-openresty/container/nginx_openresty_entrypoint.bash]: Done. Everything looks in place.

**Comment by muawiakh** [comment]
fixes #1588

**Pull request merged**